gdx/src/com/badlogic/gdx/math/Octree.java

```diff
@@ -0,0 +1,307 @@
+        root.getBoundingBox(boxes);
+        return boxes;
+    }
+
+    protected class OctreeNode {
+
+        protected int level;
+
+        // Represents the boundary of the node
+        private BoundingBox bounds;
+
+        private Array<OctreeNode> children;
+        private ObjectSet<T> geometries = null;
+
+        public OctreeNode (Vector3 min, Vector3 max, int level) {
+            this(min.x, min.y, min.z, max.x, max.y, max.z, level);
+        }
+
+        public OctreeNode (float x1, float y1, float z1, float x2, float y2, float z2, int level) {
+            bounds = new BoundingBox(new Vector3(x1, y1, z1), new Vector3(x2, y2, z2));
+            this.level = level;
+        }
+
+        protected void split (int maxItemsPerNode, Collider<T> collider) {
+            children = new Array<>(8);
+
+            float midx = (bounds.max.x + bounds.min.x) * 0.5f;
+            float midy = (bounds.max.y + bounds.min.y) * 0.5f;
+            float midz = (bounds.max.z + bounds.min.z) * 0.5f;
+
+            int deeperLevel = level - 1;
+
+            children.add(new OctreeNode(bounds.min.x, midy, midz, midx, bounds.max.y, bounds.max.z, deeperLevel));
+            children.add(new OctreeNode(midx, midy, midz, bounds.max.x, bounds.max.y, bounds.max.z, deeperLevel));
+            children.add(new OctreeNode(midx, midy, bounds.min.z, bounds.max.x, bounds.max.y, midz, deeperLevel));
+            children.add(new OctreeNode(bounds.min.x, midy, bounds.min.z, midx, bounds.max.y, midz, deeperLevel));
+            children.add(new OctreeNode(bounds.min.x, bounds.min.y, midz, midx, midy, bounds.max.z, deeperLevel));
+            children.add(new OctreeNode(midx, bounds.min.y, midz, bounds.max.x, midy, bounds.max.z, deeperLevel));
+            children.add(new OctreeNode(midx, bounds.min.y, bounds.min.z, bounds.max.x, midy, midz, deeperLevel));
+            children.add(new OctreeNode(bounds.min.x, bounds.min.y, bounds.min.z, midx, midy, midz, deeperLevel));
+
+            ObjectSet<T> geometries = this.geometries;
+            this.geometries = null;
+            // Move geometries from parent to children
+            for (OctreeNode child : children) {
+                for (T geometry : geometries) {
+                    child.add(geometry, maxItemsPerNode, collider);
+                }
+            }
+        }
+
+        public boolean contains (Collider<T> boundingBoxHandler, T geometry) {
+            return boundingBoxHandler.intersects(bounds, geometry);
+        }
+
+        public void add (T geometry, int maxItemsPerNode, Collider<T> collider) {
+            // If is not leaf, check children
+            if (!isLeaf()) {
+                for (OctreeNode child : children) {
+                    child.add(geometry, maxItemsPerNode, collider);
+                }
+                return;
+            }
+
+            if (!collider.intersects(bounds, geometry)) {
+                return;
+            }
+
```

Comment:
if/else statement could be more readable and cleanr than returning in the middle of the process